pkg/networkservice/common/connect/server.go

```diff
@@ -44,11 +45,17 @@ type connectServer struct {
 //             connect presumes depends on some previous chain element having set clienturl.WithClientURL so it can know
 //             which client to address.
 func NewServer(clientFactory func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient) networkservice.NetworkServiceServer {
+	return NewServerWithDialOptions(clientFactory, func() []grpc.DialOption { return nil })
+}
+
+// NewServerWithDialOptions - returns a new connect Server with specific client dial options
+func NewServerWithDialOptions(clientFactory func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient, dialOptionsSupplier func() []grpc.DialOption) networkservice.NetworkServiceServer {
```

Comment:
Why not just change NewServer to take a dialOptionsSupplier or better yet a ...grpc.DialOption ?

Reply:
fixed, thanks